vision/use-cases/video-thumbnail-generation/video_thumbnail_generation.ipynb

```diff
@@ -0,0 +1,717 @@
+      "metadata": {
+        "id": "d3df3e7686dd"
+      },
+      "outputs": [],
+      "source": [
+        "video_uri = \"gs://sample-videofile/sample_video_google_trips.webm\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "e332de1fb5df"
+      },
+      "source": [
+        "### Creating a local copy of the video for easy frame extraction"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "a1a71d47c5c1"
+      },
+      "outputs": [],
+      "source": [
+        "!gsutil cp {video_uri} sample_video.webm"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "125906be5cfa"
+      },
+      "source": [
+        "### Creating a MoviePy Clip Object (Helps in extracting frame at a given timestamp)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "931570104d40"
+      },
+      "outputs": [],
+      "source": [
+        "clip = VideoFileClip(\"sample_video.webm\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "7f9fd87f263e"
+      },
+      "source": [
+        "### Setting up Gemini 1.5 Pro API for thumbnail generation"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7ec946956d45"
+      },
+      "outputs": [],
+      "source": [
+        "# define fixed schema for Gemini outputs\n",
+        "response_schema = {\n",
+        "    \"type\": \"array\",\n",
+        "    \"items\": {\n",
+        "        \"type\": \"object\",\n",
+        "        \"properties\": {\n",
+        "            \"timestamp\": {\n",
+        "                \"type\": \"string\",\n",
+        "            },\n",
+        "            \"caption\": {\n",
+        "                \"type\": \"string\",\n",
+        "            },\n",
+        "        },\n",
+        "        \"required\": [\"timestamp\", \"caption\"],\n",
+        "    },\n",
+        "}"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "e9929f7007a6"
+      },
+      "outputs": [],
+      "source": [
+        "generation_config = GenerationConfig(\n",
+        "    temperature=1,\n",
+        "    top_p=0.8,\n",
+        "    max_output_tokens=8192,\n",
+        "    response_schema=response_schema,\n",
+        "    response_mime_type=\"application/json\",\n",
+        ")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "fd2aea00f91f"
+      },
+      "source": [
+        "### Define a function to Call Gemini API"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "bf07e42bf777"
+      },
+      "outputs": [],
+      "source": [
+        "def call_gemini_1_5_pro(\n",
```

Comment:
I know you just re-named this. But it should probably be more generic like `call_gemini()` Since the models are being updated frequently and the code doesn't need to change.

Reply:
Done